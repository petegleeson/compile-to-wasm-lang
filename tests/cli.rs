use assert_cmd::prelude::*; // Add methods on commands
use std::io::Write;
use std::process::Command; // Run programs
use tempfile::Builder;

#[test]
fn global_declarations() -> Result<(), std::io::Error> {
  let mut cmd = Command::cargo_bin("compile-to-wasm-rust").expect("unable to find binary");
  let src = r#"
    let x = 1;
    let y = () {
      1;
    };
  "#;
  let mut file = Builder::new()
    .prefix("global-declarations")
    .rand_bytes(0)
    .tempfile()?;
  writeln!(file, "{}", src)?;
  cmd.arg("--ir").arg(file.path()).output().map(|res| {
    insta::assert_snapshot!(std::str::from_utf8(&res.stdout).unwrap());
  })
}
