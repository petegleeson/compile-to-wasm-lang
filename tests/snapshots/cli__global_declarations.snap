---
source: tests/cli.rs
expression: "std::str::from_utf8(&res.stdout).unwrap()"
---
; ModuleID = 'global-declarations'
source_filename = "global-declarations"

@x = global i32 1
@y = global i32 ()* @fn_8

define i32 @fn_8() {
entry:
  ret i32 1
}

âœ… Done

